topology/src/main/java/io/camunda/zeebe/topology/state/ClusterTopology.java

```diff
@@ -89,9 +103,51 @@ public ClusterTopology merge(final ClusterTopology other) {
               .collect(
                   Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue, MemberState::merge));
 
-      // TODO: changes also have to be merged. We will do it when we add support for configuration
-      // changes.
-      return new ClusterTopology(version, ImmutableMap.copyOf(mergedMembers), changes);
+      final var mergedChanges = changes.merge(other.changes);
+      return new ClusterTopology(version, ImmutableMap.copyOf(mergedMembers), mergedChanges);
     }
   }
+
+  /**
+   * @return true if the next operation in pending changes is applicable for the given memberId,
+   *     otherwise returns false.
+   */
+  public boolean hasPendingChangesFor(final MemberId memberId) {
+    return !changes.pendingOperations().isEmpty()
+        && changes.pendingOperations().get(0).memberId().equals(memberId);
+  }
+
+  /**
+   * Returns the next pending operation for the given memberId. If there is no pending operation for
+   * the members, throws NoSuchElementException. It is recommended to call {@link
+   * #hasPendingChangesFor(MemberId)} before calling this method.
+   *
+   * @param memberId id of the member
+   * @return the next pending operation for the given memberId.
+   * @throws NoSuchElementException if there is no pending operation for the given memberId.
+   */
+  public TopologyChangeOperation pendingChangerFor(final MemberId memberId) {
+    if (!hasPendingChangesFor(memberId)) {
+      throw new NoSuchElementException();
+    }
+    return changes.pendingOperations().get(0);
+  }
+
+  /**
+   * When the operation returned by {@link #pendingChangerFor(MemberId)} is completed, the changes
+   * should be reflected in ClusterTopology by invoking this method. This removes the completed
+   * operation from the pending changes and update the member state using the given updater.
+   *
+   * @param memberId id of the member which completed the operation
+   * @param memberStateUpdater the method to update the member state
+   * @return the updated ClusterTopology
+   */
+  public ClusterTopology advanceTopologyChange(
+      final MemberId memberId, final UnaryOperator<MemberState> memberStateUpdater) {
+    return updateMember(memberId, memberStateUpdater).advance();
+  }
+
+  private ClusterTopology advance() {
+    return new ClusterTopology(version, members, changes.advance());
```

Comment:
Version of the ClusterTopology needs to be increased only when starting a new configuration change. For every other change, `MemberState` and `ClusterChangePlan` updates their own versions to help merging. Eventhough, we are currently only applying one operation at a time, it is possible to support parallel operations in future. In that case, if multiple members update `ClusterTopology.version`, it would result in conflict. So it is better to only allow the coordinator to update the version, and each member to only update their own MemberState.version.